internal/provisioner/helm_utils.go

```diff
@@ -283,16 +289,66 @@ func (d *helmDeployment) Version() (*model.HelmUtilityVersion, error) {
 	return nil, errors.Errorf("unable to get version for chart %s", d.chartDeploymentName)
 }
 
-func applyGitlabTokenIfPresent(original string) string {
-	if os.Getenv(model.GitlabOAuthTokenKey) == "" {
-		return original
+type gitlabValuesFileResponse struct {
+	Content string `json:"content"`
+}
+
+const temporaryValuesFile string = "/tmp/helm-values.yaml"
```

Comment:
Just checking, this works because we never run two different threads that invoke helm at the same time right? I think that is a safe assumption to make for now, but just wanted to double-check my assumption.

Reply:
I was going to ask you to confirm that 😅

Reply:
Since neither of seem certain about the safety of using the same static file over and over I changed this to use the standard library temp file creator and added a `defer` to delete it in the calling function after the operation is complete @gabrieljackson

Reply:
Yeah, I think this is much better and we won't have to worry about file change collisions anymore which is great.